### ISSUE TITLE
Dapr fails to pull image on Windows 2022 Node Pool in Azure Kubernetes Service

### ISSUE THREAD

**Issue by joshuarmclean** [dapr/dapr#5644]
## In what area(s)?

/area placement

> /area docs

> /area test-and-release

## What version of Dapr?

1.8.3

## Expected Behavior

When we deploy our Windows 2022 pods to a Windows 2022 node pool in AKS it will run.

## Actual Behavior

When we deploy our 2022 container onto a 2022 node pool we get the following error:
Normal   BackOff    24s                kubelet            Back-off pulling image "ghcr.io/dapr/daprd:1.8.3"
Dapr support had me switch where containers are pulled form.

Initial Errors were:
35m         Normal    Pulling                   pod/pvsbx-5c5f55998f-nhrs9           Pulling image "docker.io/daprio/daprd:1.8.3"
35m         Warning   Failed                    pod/pvsbx-5c5f55998f-nhrs9           Error: ErrImagePull
99s         Normal    BackOff                   pod/pvsbx-5c5f55998f-nhrs9           Back-off pulling image "docker.io/daprio/daprd:1.8.3"

We spoke with dapr support on discord and were directed to report the issue on github.

## Steps to Reproduce the Problem
Using Dapr 1.8.3 with Azure Kubernetes 1.23.12 create a Windows 2022 node pool which has to be done via az cli:
https://learn.microsoft.com/en-us/azure/aks/learn/quick-windows-container-deploy-cli
az aks nodepool add \
    --resource-group myResourceGroup \
    --cluster-name myAKSCluster \
    --os-type Windows \
    --os-sku Windows2022 \ 
    --name npwin \
    --node-count 1

Then try to deploy a windows 2022 container to the Windows 2022 node pool.
NOTE:
If you modify the pod annotation:
from:
dapr.io/enabled: "true"
to 
dapr.io/enabled: "false"
The pod will deploy.  But swithcing back to dapr.io/enabled: "true" the pod will fail to pull the dapr image.

See above.
## Release Note

RELEASE NOTE:

**Comment by artursouza** [comment]
I installed a brand new VM with Windows Server 2022 Standard Edition, enabled Windows Containers, installed Docker and ran docker pull. It worked. Can you provide more details on your error?

![image](https://user-images.githubusercontent.com/130954/207992584-6a66419d-01b6-46be-85e8-7e7b7d685f1a.png)

**Comment by joshuarmclean** [comment]
We are running Windows 2022 Server on nodes in our our Node Pools in Azure Kubernetes.We cannot run our deployment yaml, it throws the error I provided that it cannot download the dapr image due to an OS mismatch.When I set the dapr annotation to false for enabled our Windows 2022 container starts.Can you run an Windows 2022 container in AKS on an Windows 2022 node pool using dapr 1.8.3 as a side car?I don’t know that what you are doing is a correct reproduction of our issue.  Can you let me know what more details you would like?I can provide screenshots, screen share etc.Thank you for the help.Sent from my iPhoneOn Dec 15, 2022, at 6:08 PM, Artur Souza ***@***.***> wrote:﻿
I installed a brand new VM with Windows Server 2022 Standard Edition, enabled Windows Containers, installed Docker and ran docker pull. It worked. Can you provide more details on your error?

**Comment by yaron2** [comment]
Have you read this? https://docs.dapr.io/operations/hosting/kubernetes/kubernetes-hybrid-clusters/

**Comment by joshuarmclean** [comment]
Yes I have read that.
To be clear, we are currently running Windows 2019 Containers on Windows
2019 node pools in AKS using dapr 1.8.3.
Everything has been working fine for about 1 year.

Our issue started when we saw that as of AKS 1.23.0 Windows 2022 was
supported in AKS and we wanted to move to Windows 2022 for the smaller
container image size.

We rebuilt our container on 2022, and setup a node pool using the
documentation here:
https://learn.microsoft.com/en-us/azure/aks/upgrade-windows-2019-2022

While it also mentions the node affinity option it made no difference for
us.  We used a taint for our node pool to ensure the new 2022 pods landed
in the 2022 node pool.
Our pod was NOT able to pull the dapr image and it threw the error about a
host os mismatch.

When I redeployed I modified the dapr annotation from:
dapr.io/enabled: "true"
to
dapr.io/enabled: "fale"

When I did this I was able to deploy my 2022 container on the 2022 node
pool without dapr.

I should mention that Windows 2019 containers are not compatible with
Windows 2022.  So you have to run WIndows 2019 containers on Windows 2019
node pools and you have to run Windows 2022 containers on WIndows 2022 node
pools.

This is why I have raised the issue.   I think there is an issue with
running the dapr image on an Windows 2022 Node Pool.

Let me know if you need any further information.

Thank you for the help.
Josh

**Comment by shubham1172** [comment]
I can reproduce this error.

<img width="1728" alt="image" src="https://user-images.githubusercontent.com/22132944/208581731-e9610bed-7486-4609-9607-3beba894b082.png">

Installed Dapr:

```bash
helm install dapr dapr/dapr --namespace dapr-system --set global.daprControlPlaneOs=windows --create-namespace
```

Pods are not able to pull the image.

```bash
k get pods -n dapr-system
```

<img width="918" alt="image" src="https://user-images.githubusercontent.com/22132944/208582442-73da0a5e-c11f-4157-8acf-d87ef42a9715.png">


Looking into k8s events -

```bash
kubectl get events --all-namespaces --sort-by='.metadata.creationTimestamp'
```

`Failed to pull image "docker.io/daprio/dapr:1.9.5": rpc error: code = NotFound desc = failed to pull and unpack image "docker.io/daprio/dapr:1.9.5": no match for platform in manifest: not found`

**Comment by 1046102779** [comment]
I found dapr version 1.9.5 for windows/amd64.

<img width="1245" alt="image" src="https://user-images.githubusercontent.com/20457624/208583028-2e8dc54c-7c4a-4d9c-ad64-5fed14339fdb.png">

**Comment by shubham1172** [comment]
> I found dapr version 1.9.5 for windows/amd64.
> 
> <img alt="image" width="1245" src="https://user-images.githubusercontent.com/20457624/208583028-2e8dc54c-7c4a-4d9c-ad64-5fed14339fdb.png">

It is there indeed, but note that we use mcr.microsoft.com/windows/nanocore 1809, which might not be compatible with the Windows 2022 nodes. It **might** have something to do with https://github.com/containerd/containerd/issues/6508

**Comment by shubham1172** [comment]
## TLDR;

On updating the Windows [servercore](https://mcr.microsoft.com/en-us/product/windows/servercore/tags) and [nanoserver](https://mcr.microsoft.com/en-us/product/windows/nanoserver/tags) images from `1809` to `ltsc2022`

Based on https://learn.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/version-compatibility, they recommend either to use a matching version based on the host (in this case using the ltsc2022 image for Windows Server 2022 nodes), or using Hyper-V (based on a specific matrix), but the latter is always not available.

A possible mitigation is to start supporting the ltsc2022 image alongside our 1809 image.

## Summary

Steps taken briefly-
```bash
# Create a new base image
docker build -f ./docker/Dockerfile-windows-base ./docker -t ghcr.io/shubham1172/windows-base:ltsc2022

# Build the dapr image
docker build --build-arg PKG_FILES=* -f .\docker\Dockerfile-windows ./dist/windows/release -t ghcr.io/shubham1172/dapr:ltsc2022

# Push the image
docker push ghcr.io/shubham1172/dapr:ltsc2022
```

```bash
helm install dapr dapr/dapr --namespace dapr-system --set global.daprControlPlaneOs=windows --set global.registry=ghcr.io/shubham1172 --set global.tag=ltsc2022
```

![image](https://user-images.githubusercontent.com/22132944/208875878-ea08dca5-0549-41a3-ab5b-46dfc66df15f.png)

**Comment by mschumacher-syntellis** [comment]
What is status on this?

**Comment by joshuarmclean** [comment]
Was there an update on this so we can use dapr on Windows 2022?

**Comment by mschumacher-syntellis** [comment]
> Was there an update on this so we can use dapr on Windows 2022?
Us too!

**Comment by riverar** [comment]
Am seeing `no match for platform in manifest: not found` for `docker.io/daprio/daprd:latest` on Windows Server 2022. What's the status of using Dapr sidecars on this OS? Heads up: This is the current LTSC and has been out for two years now.

**Comment by riverar** [comment]
@shubham1172 Perfect thanks for the lightning response, I should have looked at the actual manifest.
